api/game_configurations/loops.yaml

```diff
@@ -5,74 +5,147 @@ status: active
       - output
       - input
-  - external_id: while_loops_1
-    title: Advnaced while loops
-    description: Write some advnaced while loops
-    prompt: placebolder
+      - math_number
+      - controls_if
+      - controls_whileUntil # while loop
+      - controls_for # for loop
+      - controls_flow_statements # breakout
+      - math_number_property # even
+      - math_arithmetic # add, subtract, multiply ..
+  - external_id: while_loop
+    title: While - advanced for-loop
+    description: Print sum of integers from 1 to up until the given integer
     status: active
     level_number: 3
-    inputs: int
-    num_inputs: 10
+    inputs: pos_int
+    num_inputs: 1
+    prompt: >
+      You can practice a <strong>while</strong>-loop and (counter) variable with this blockly game. <br>
+      <cite>In most computer programming languages, a <strong>while</strong> loop is a control flow statement that allows code to be executed repeatedly based on a given Boolean condition. 
+      The while loop can be thought of as a repeating if statement. (Wikipedia) </cite>
+      <br>
+      It is possible to express more dynamic logics with this while loop because a while loop has more powerful looping behaviour than for-loops.<br>
+      To execute a <strong>while</strong> statement, the logic first evaluates expression or condition.
+      The loop keeps continuing until the condition is true.
+      However, if the condition is false, the flow skips over the loop body and moves on to the next statement.
+      <li>Instruction</li> 
+      You will be given a positive integer <strong>N</strong>. 
+      Print the sum of positive integers from 1 to the given integer. <br>
+      sum = 1 + 2 + 3 + ... + N
     blocks:
+      - variable
       - text
       - output
       - input
-  - external_id: nested_loops
-    title: Nested Loops
-    description: Write loops within loops
-    prompt: placebolder
+      - math_number
+      - math_arithmetic
+      - logic_compare
+      - controls_whileUntil
+      - controls_for
+  - external_id: while_decimal
+    title: While - decimal to binary
+    description: Given a passive integer input, print out a specific pattern
     status: active
     level_number: 4
-    inputs: int
-    num_inputs: 10
+    inputs: pos_int
+    num_inputs: 1
+    prompt: >
+      <cite> Binary is a base 2 number system. <br>
+      Base 2 means there are only two digits—1 and 0—which correspond to the on and off states your computer can understand.
+      You’re probably familiar with base 10—the decimal system. Decimal makes use of ten digits that range from 0 to 9, 
+      and then wraps around to form two-digit numbers, with each digit being worth ten times more than the last (1, 10, 100, etc.).
+      Binary is similar, with each digit being worth two times more than the last. <strong>(howtogeek.com)</strong><br>
+      <li>Instruction</li> 
+      Given a positive number between 0 < <strong>P</strong> < 32 <br>
```

Comment:
it could be worth giving them a bit more of a drawn out example. 

"for example, the number 13 can be thought of as 1*10^1 + 3*10^0. Similarly the number 13 can also be thought of as "1*2^3 + 1*2^2 + 0*2^1 + 1*2^0 - or the number 1101 in binary".

Reply:
Followed up.